site/content/docs/getting-further/is-my-app-testable.md

```diff
@@ -0,0 +1,94 @@
+
+ZAP can help you secure your web apps, but you can make life much more difficult for ZAP, and that in turn makes life more difficult for you.
+
+Making it harder for ZAP to test your web apps will not make them more secure, it can in fact make them less secure.
+If you make automated testing too hard then the race to find vulnerabilities in your apps will be between
+the pentesters you hire and the malicious attackers who are out to compromise you.
+
+### Provisioning
+
+Can you set up your app in a test environment with no manual input?
+
+This includes provisioning all of the services, configuring them to work together, and creating a good range of test data, and provisioning test users for all valid roles.
+Your app should not depend on any third party services that cannot be mocked, or interact with any other services which are not part of the isolated test environment.
+
+If you cannot set easily set up your app in a reliable state in a dedicated test environment then the reliablility of the results you get will vary.
+
+### Performance
+
+If your test environment is slow under normal circumstances then it will really struggle when scanned by ZAP.
+ZAP may fail to discover content or miss vulnerabilities that it would otherwise find due to requests timing out.
+
+You can configure ZAP to handle apps that perform badly, but that is a manual process which you may not realise you need to perform.
+
+### Authentication
+Can automated tools authenticate to your application easily?
+
+Complex login procedures make it harder for ZAP to authenticate to your app. If ZAP cannot login to your app then it cannot test it effectrively.
+
+ZAP can handle pretty much any form of authentication, but the more complex it is the more manual configuration will be required.
+
+### Exploration
+Can ZAP explore your app effectively?
+
+If your web app is designed for humans then you will hopefully have designed it in a way that makes it clear to humans how to navigate your app.
+That is not the same as making your app easy to navigate by automated tools.
+
+HTML links are easy to identify. Onclick handlers on elements like DIVs are much harder to identify.
+
+The [AJAX Spider](/docs/desktop/addons/ajax-spider/) can struggle with popup menus, although the 
+[Client Spider](/docs/desktop/addons/client-side-integration/spider/) can handle these in many cases.
+
+None of the ZAP spiders can curerently handle Shadow DOMs.
+
+If you have integration tests for you app then you should look at [proxying](/docs/getting-further/automation/exploring-your-app/#proxying-integration-tests)
+then through ZAP as these can significantly increase coverage.
+
+### API Definitions
+Do you provide an API definition?
+
+If your app provides an API then you should also maintain an API definition that can be used by automated tools. This definition does not have to be public, but it should be generated automatically.
```

Comment:
Do we need to state or care how it's generated?

Reply:
I dont think so

Reply:
:wink: Then let's not :wink:

> This definition does not have to be public, but it should be generated automatically.